### ISSUE TITLE
eslint-disable-next-line fails

### ISSUE THREAD

**Issue by tbillington** [eslint/eslint#6869]
**What version of ESLint are you using?**
`v3.2.2`
**What parser (default, Babel-ESLint, etc.) are you using?**
Whichever the latest airbnb config uses.
**Please show your full configuration:**
```json
{
  "extends": "airbnb"
}
```
**What did you do? Please include the actual source code causing the issue.**
```javascript
render() {
    // eslint-disable-next-line
    const {name, data, namespace, ...others} = this.props; // Linter errors here
```
**What did you expect to happen?**
The lint error to be hidden
**What actually happened? Please include the actual, raw output from ESLint.**
The lint error was not hidden
```bash
18:35  error  Parsing error: Unexpected token ..

✖ 1 problem (1 error, 0 warnings)
```

**Comment by platinumazure** [comment]
Per our [configuration docs](http://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments), ESLint still has to parse the entire file, so all code in a "disabled" line must still parse.

Please show your ESLint config and we can help you figure out what's wrong. At minimum you need parserOptions ecmaVersion at 6 or higher, and you may also need ecmaFeatures: experimentalObjectRestSpread: true as well, but I'm not 100% sure.

**Comment by mysticatea** [comment]
Yes. Comments to disable rules cannot disable parsing errors. The parsing errors is not from any rule.

**Comment by tbillington** [comment]
The airbnb config has `emcaVersion: 6` set, is that not enough to cover the spread operator ?

**Comment by vitorbal** [comment]
@tbillington to enable rest/spread for objects, you will need to turn on the `experimentalObjectRestSpread` [parser option](http://eslint.org/docs/user-guide/configuring#specifying-parser-options).

**Comment by mysticatea** [comment]
@tbillington The spread operators are not a part of JavaScript yet. It's [a stage 2 proposal](https://github.com/tc39/proposals#ecmascript). So you require additional flag (`experimentalObjectRestSpread`) in order to use the experimental syntax.

**Comment by tbillington** [comment]
Ah, I see. Thank you.

**Comment by vitorbal** [comment]
Closing, as the issue seems to be resolved. Please reopen if there is more to discuss.
